core/query_factory.py

```
@@ -269,3 +269,115 @@ def _make_for_test(self):
+        NO_KEY_WITH_DIST = tr("All OSM objects in {dist} meters of {extent} are going to be downloaded")
+
+        ATTRIB_ONLY = tr("All OSM objects with the key {key} are going to be downloaded")
+        
+        extent_lbl = ""
+        dist_lbl = ""
+        key_lbl = ""
+        use_with_dist = False
+        use_all_vals = False
+        attrib_only = False
+        
+        # first translate the location information
+        have_loc_info = False
+        if self._query_type == QueryType.InArea:  #'in':
+            if place is not None:
+                extent_lbl = place
+                have_loc_info = True
+                
+        elif self._query_type == QueryType.AroundArea:  #'around':
+            if place is not None:
+                extent_lbl = place
+                dist_lbl = str(self._distance_around)
+                use_with_dist = True
+                have_loc_info = True
+
+        elif self._query_type == QueryType.BBox:  # 'canvas' or 'layer'
+            extent_lbl = tr("the canvas or layer extent")
+            have_loc_info = True
+            
+        elif self._query_type == QueryType.NotSpatial:  #'attributes':
+            geomsg = ""
+            have_loc_info = True
+            attrib_only = True
+        else:
+            have_loc_info = False
+
+        if not have_loc_info:
+            msg = tr("Please select an input location or area.")
+            return msg
+
+        # Next get the key / values
+        if len(self._key)==0:
+            key = None
+        else:
+            key = self._key[0]
+
+        if len(self._value)==0:
+            val = None
+        else:
+            val = self._value[0]
+                
+        if key is None:
+            if val is not None:
+                return tr("You have specified a value without a matching key.")
+
+            elif attrib_only:
+                return tr("You have must specify a key.")
```

Same L352 and L355, this should be handled in the `check_parameters`.

---

I think the one on line 355 can still happen?
(I saw your earlier message and started to remove these extraneous checks before see these line by line comments - agree on all the others)

---

> I think the one on line 355 can still happen?

if it's the case, it's a mistake from `check_parameters`. It should raise the error, it's the purpose of `check_parameters`. Do you want to check or to create a ticket?

---

After calling `check_parameters`, you must feel safe about the query and its parameters.

---

Yes, just confirmed. I'll file an issue. If you press Run Query it then gives an Overpass error about the filters being too weak (just a bit!). 
There's an erroneous word in the message above, I'll fix that check it in, and I think we're done.